TripalCultivate/TripalCultivate-Phenotypes
# G5.125 support d11

**Issue #125**

## Motivation



We need to update the automated testing grid to test these versions and fix any issues that crop up. We also need to add these versions to our automated docker builds. It may be worth separating our current image into a base and module just like I did in Tripal core for faster build times and easier testing.

## What does this PR do?


- [x] Update GitHub workflows to include Drupal 10.5, 11.0, + 11.1 and remove 10.2.
- [x] Update devcontainer to run on Drupal 11.1 and to use the new base image.

## Testing

### Automated Testing


This PR does not add any additional automated tests. Instead it simply expands the versions these tests are ran on. 

### Manual Testing


1. Checkout this branch locally, open it in VSCode and "Reopen in Container".
2. Confirm that devcontainer builds properly (i.e. no screaming error 😉) and open the webpage in the browser to confirm versions. Note it is now building Drupal 11.1
<img width="1178" alt="Screenshot 2024-11-21 at 4 10 40 PM" src="https://github.com/user-attachments/assets/8d892f07-803b-478a-8bf8-48ea1dbabdc9">

3. Also check the README on this branch by going to the following link and confirm the testing grid renders properly with it's badges. 
https://github.com/TripalCultivate/TripalCultivate-Phenotypes/tree/g5.125-supportD11?tab=readme-ov-file#automated-testing
<img width="1114" alt="Screenshot 2024-11-21 at 4 49 52 PM" src="https://github.com/user-attachments/assets/9f7fb691-a226-4afc-9160-832f0acca688">


4. Configure the module for use and confirm no errors pop-up while you do so:
    a) Create an organism
    b) Go to configure ontologies (admin/tripal/extension/tripal-cultivate/phenotypes/ontology) and configure the organism
    c) Go to the trait importer and ensure it looks right
